### ISSUE TITLE
Updating the change log.

### ISSUE THREAD

**Issue by DigiLive** [JBlond/php-diff#66]
Both of us forgot the change log.

_Originally posted by @JBlond in https://github.com/JBlond/php-diff/issue_comments/707764891_

**Comment by DigiLive** [comment]
Can something like this be implemented?
https://github.com/marketplace/actions/generate-changelog

**Comment by JBlond** [comment]
Yes we can. I have zero experience it in.

**Comment by DigiLive** [comment]
Haha... I don't have any experience with GitHub actions either.

It's gonna be a trial and error, I guess.

**Comment by JBlond** [comment]
I made a script for that (WIP). It can only track the changes after tagging so far. 

See https://github.com/JBlond/php-diff/blob/automatic-changelog/changelog.sh
and
https://github.com/JBlond/php-diff/blob/automatic-changelog/changelog.md

maybe you have some additional ideas for that.

**Comment by DigiLive** [comment]
Looks promising so far.
My suggestions are:
- Don't include merge commits like: merge master into development
- Remove duplicate commit titles per release. (E.g. `Fix #58 - Side by side diff shows empty diff` is mentioned 3x)
- Include commits which where made after the latest release.
- Include links to Issues. (E.g. Fix #58 - Side by side diff shows empty diff)

I'm thinking about what's more sucker proof...
- Do we need to run the script manually (and won't we accidentally forget)?
- Assuming the script runs locally and because of cross-platform development, it might not be handy to have a single-platform script.

These points are inapplicable when running GitHub actions, I think.

**Comment by DigiLive** [comment]
Investigating GitHub actions, I came up with the following:
https://github.com/DigiLive/testChangelog/blob/master/.github/workflows/updateChangeLog.yml
(which operates on push events on the master branch for testing)

It uses https://github.com/github-changelog-generator/github-changelog-generator,
but I'm not sure when, what's included into the change log.
I think it depends on issues, PR's and their states and labels.

**Comment by DigiLive** [comment]
Please give https://github.com/DigiLive/gitChangelog a try.

**Comment by JBlond** [comment]
I made a branch for that https://github.com/JBlond/php-diff/tree/66-automatic-changelog

**Comment by DigiLive** [comment]
👍🏻
Please be aware the library is in an unstable state as long as the version is below 1.0.0.
Until then breaking changes may apply to future updates.

**Comment by JBlond** [comment]
it is not merged yet :D

**Comment by DigiLive** [comment]
I know... Just a warning. :)

For testing you could pull the latest commit of the main branch with composer.

If you have any questions about setting up or using the lib, or have any ideas, contact me or create a new issue at the repo.

**Comment by DigiLive** [comment]
RC1 for the GitChangelog package is released.
Please grab it at https://github.com/DigiLive/gitChangelog/releases

I don't expect the code will change between this release and the first major release which is scheduled for 15-12-2020. 
Report any bugs/comments asap before this date.

**Comment by JBlond** [comment]
maybe upgrading this lib's requirement to 7.3 is not needed. The Changelog would only be needed for development.

**Comment by DigiLive** [comment]
Then (I think) you won't be able to install GitChanglog with composer.json of this lib, unless the version constraint of GitChanglog is lowered.

Possible solutions:
* Don't include GitChangelog to this project with composer.json, but install it separately.
  (It should be able to process git repositories in other local folders.)

* Update GitChangelog to the same version constraint as this project.
  (The library code should be compatible, PHPunit tests may be incompatible.)

Please feedback what would work best for you.

**Comment by JBlond** [comment]
I gonna upgrade this lib requirement to PHP 7.3

**Comment by DigiLive** [comment]
v1.0.0 of DigiLive/gitChangelog is released since today.

**Comment by JBlond** [comment]
Once https://github.com/JBlond/php-diff/pull/85 is merged, we can start with this.

**Comment by JBlond** [comment]
There is now #89

**Comment by JBlond** [comment]
This has been done in https://github.com/JBlond/php-diff/pull/89
